capstone-server/src/main/java/com/hcl/capstoneserver/invoice/InvoiceService.java

```diff
@@ -0,0 +1,55 @@
+package com.hcl.capstoneserver.invoice;
+
+import com.hcl.capstoneserver.invoice.dto.InvoiceDTO;
+import com.hcl.capstoneserver.invoice.entities.Invoice;
+import com.hcl.capstoneserver.invoice.exceptions.InvoiceNumberExistsException;
+import com.hcl.capstoneserver.invoice.repositories.InvoiceRepository;
+import com.hcl.capstoneserver.user.UserService;
+import com.hcl.capstoneserver.user.entities.Client;
+import com.hcl.capstoneserver.user.entities.Supplier;
+import org.modelmapper.ModelMapper;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+import java.util.Optional;
+
+@Service
+public class InvoiceService {
+
+    private final InvoiceRepository invoiceRepository;
+    private final ModelMapper mapper;
+    private final UserService userService;
+
+    public InvoiceService(InvoiceRepository invoiceRepository, ModelMapper mapper, UserService userService) {
+        this.invoiceRepository = invoiceRepository;
+        this.mapper = mapper;
+        this.userService = userService;
+    }
+
+    public Invoice createInvoice(InvoiceDTO invoice) {
+        try {
+            Optional<Client> client = userService.fetchClientById(invoice.getClientId());
+            Optional<Supplier> supplier = userService.fetchSupplierById(invoice.getSupplierId());
+
+            if (!client.isPresent() || !supplier.isPresent()) {
```

Comment:
need to change this, only clients can access this route. so we only have to check whether the supplier exists or not